### DISCUSSION TITLE
How to implemnt user authentication and modify rest API

### DISCUSSION THREAD

**Discussion by lantianjialiang** [usnistgov/NEMO#110] [Q&A] [answered]
Hi Dear NEMO

I want to use NEMO in our product environment, but there have two problems which we consider.

1. There are only have LDAP and Kerberos authetication
2. There not have modify REST API, juts view REST API
What we should do if we want implemnt in NAMO?

Could you please give me some hints and document to start?
If we implement those feature, can we make a Pull Request to you repo?

Sincerely
Liang

**Comment by rptmat57** [comment, marked as answer]
Hi Liang,

For 1. you can use the [NEMO-allauth plugin](https://gitlab.com/nemo-community/prometheus-computing/nemo-allauth) which provides a lot more options for authentication. It is based on django-allauth and support a ton of [providers](https://django-allauth.readthedocs.io/en/latest/providers.html).

For 2. you are correct, at the moment NEMO's REST API only supports data read. If you want to implement POST, PUT, DELETE etc. for NEMO you can either submit a PR, or create your own plugin to add this functionality.
I would have to check with the other stakeholders, but I think PRs to add more REST support would be welcomed.

Thanks,
Mathieu

**Comment by lantianjialiang** [comment]
Hi Mathieu

Got it, thanks for your response. we are evaluating NEMO now, if we decide to contribute to it, we will let you know.

Sincerely
LiangK

**Comment by rptmat57** [comment]
I just wanted to add that we recently expanded the REST API to allow POST, PUT, partial updates (PATCH), and DELETE.
